kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h

```diff
@@ -513,54 +517,71 @@ class ResidualBasedBlockBuilderAndSolver
             set_type dofs_tmp_set;
             dofs_tmp_set.reserve(20000);
 
+            // Gets the array of elements from the modeler
             #pragma omp for schedule(guided, 512) nowait
-            for (int i = 0; i < nelements; i++)
-            {
-                typename ElementsArrayType::iterator it = pElements.begin() + i;
+            for (int i = 0; i < number_of_elements; ++i) {
+                auto it_elem = r_elements_array.begin() + i;
 
-                // gets list of Dof involved on every element
-                pScheme->GetElementalDofList(*(it.base()), ElementalDofList, CurrentProcessInfo);
-
-                dofs_tmp_set.insert(ElementalDofList.begin(), ElementalDofList.end());
+                // Gets list of Dof involved on every element
+                pScheme->GetElementalDofList(*(it_elem.base()), dof_list, r_current_process_info);
+                dofs_tmp_set.insert(dof_list.begin(), dof_list.end());
             }
 
-            KRATOS_INFO_IF("ResidualBasedBlockBuilderAndSolver", ( this->GetEchoLevel() > 2)) << "Initializing condition loop" << std::endl;
+            // Gets the array of conditions from the modeler
+            ConditionsArrayType& r_conditions_array = rModelPart.Conditions();
+            const int number_of_conditions = static_cast<int>(r_conditions_array.size());
+            #pragma omp for  schedule(guided, 512) nowait
+            for (int i = 0; i < number_of_conditions; ++i) {
+                auto it_cond = r_conditions_array.begin() + i;
 
-            ConditionsArrayType& pConditions = rModelPart.Conditions();
-            const int nconditions = static_cast<int>(pConditions.size());
-            #pragma omp for  schedule(guided, 512)
-            for (int i = 0; i < nconditions; i++)
-            {
-                typename ConditionsArrayType::iterator it = pConditions.begin() + i;
+                // Gets list of Dof involved on every element
+                pScheme->GetConditionDofList(*(it_cond.base()), dof_list, r_current_process_info);
+                dofs_tmp_set.insert(dof_list.begin(), dof_list.end());
+            }
 
-                // gets list of Dof involved on every element
-                pScheme->GetConditionDofList(*(it.base()), ElementalDofList, CurrentProcessInfo);
-                dofs_tmp_set.insert(ElementalDofList.begin(), ElementalDofList.end());
+            // Gets the array of constraints from the modeler
+            auto& r_constraints_array = rModelPart.MasterSlaveConstraints();
+            const int number_of_constraints = static_cast<int>(r_constraints_array.size());
+            #pragma omp for  schedule(guided, 512) nowait
+            for (int i = 0; i < number_of_constraints; ++i) {
+                auto it_const = r_constraints_array.begin() + i;
+
+                // Gets list of Dof involved on every element
+                it_const->GetDofList(dof_list, second_dof_list, r_current_process_info);

```

Comment:
just to say that instead of second_dof_list you could have named something like slave_dofs_list or smthg of the sort, just to make more clear what you do.

btw, my error that this loop was not included